ModuleManager fixes / improvements

EmotionalLove opened the pull request:
custom module names aren't known as compile-time, so if a user renamed "Freecam" to "Spectator", or something similar, the lookup map would surely map "Spectator" to the "Freecam" module, but the code still asks ModuleManager for "Freecam" regardless.

cookiedragon234 commented:
Why would people be renaming module names? If they are then they should rename all references themselves. This creates unreadable code, where you see something requesting a module "Spectator", you look for the spectator module, but then you cant find the spectator module.

EmotionalLove commented:
> Why would people be renaming module names? If they are then they should rename all references themselves. This creates unreadable code, where you see something requesting a module "Spectator", you look for the spectator module, but then you cant find the spectator module.

again, don't talk to me about this. @zeroeightysix made the modules renameable iirc

cookiedragon234 commented:
Do you mean them being renamed in the code or is there some sort of runtime mechanism for updating their names?

cookiedragon234 commented:
If its the case that people can rename them at runtime you should instead make it update the name that is displayed in the gui rather than the internal name.

S-B99 commented:
they're updated in Module.Info, config side, but for enabling modules by name it looks for literally their name

S-B99 commented:
I'm pretty sure it doesn't change the internal name?

S-B99 commented:
It is updated gui side and cli side, otherwise it's the same in code

EmotionalLove commented:
> Do you mean them being renamed in the code or is there some sort of runtime mechanism for updating their names?

renamed during runtime. you could rename the modules using a command. There's a field called `originalName` that keeps the original name, I updated `ModuleManager` to rely on that instead of the `name` field, which can change.

cookiedragon234 commented:
Why not create a method `getDisplayName()` which returns a user friendly name of the module for the arraylist and gui. Then users can update that user friendly name. This means that all references to the module will remain the same, stopping configs and string lookups from breaking.
